modules/apps/adaptive-media/adaptive-media-image-item-selector-impl/src/main/java/com/liferay/adaptive/media/image/item/selector/internal/FileEntryAMImageURLItemSelectorReturnTypeResolver.java

```diff
@@ -76,16 +73,11 @@ public String getValue(FileEntry fileEntry, ThemeDisplay themeDisplay)
 
 		JSONArray sourcesJSONArray = _jsonFactory.createJSONArray();
 
-		List<MediaQuery> mediaQueries = _mediaQueryProvider.getMediaQueries(
-			fileEntry);
-
-		Stream<MediaQuery> mediaQueryStream = mediaQueries.stream();
+		for (MediaQuery mediaQuery :
+				_mediaQueryProvider.getMediaQueries(fileEntry)) {
 
-		mediaQueryStream.map(
-			this::_getSourceJSONObject
-		).forEach(
-			sourcesJSONArray::put
-		);
+			sourcesJSONArray.put(_getSourceJSONObject(mediaQuery));
```

Comment:
@julschong Please use `JSONUtil.toJSONArray` to simplify, it's like

```
		).put(
			"sources",
			JSONUtil.toJSONArray(
				_mediaQueryProvider.getMediaQueries(fileEntry),
				this::_getSourceJSONObject)
```

Reply:
Thanks for the tip!